public/media_engagement/imaqal_media_engagement_stacked.js

```diff
@@ -0,0 +1,314 @@
+var total_received_sms_graph = d3.select(".total_received_sms_graph").append("svg")
+    .attr("width", Width + Margin.left + Margin.right)
+    .attr("height", Height + Margin.top + Margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + Margin.left + "," + Margin.top + ")");
+
+// Append total sent sms graph to svg
+var total_sent_sms_graph = d3.select(".total_sent_sms_graph").append("svg")
+    .attr("width", Width + Margin.left + Margin.right)
+    .attr("height", Height + Margin.top + Margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + Margin.left + "," + Margin.top + ")");
+
+// Append total sent sms graph to svg
+var total_failed_sms_graph = d3.select(".total_failed_sms_graph").append("svg")
+    .attr("width", Width + Margin.left + Margin.right)
+    .attr("height", Height + Margin.top + Margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + Margin.left + "," + Margin.top + ")");
+
+// Format TimeStamp  
+var timeFormat = d3.timeFormat("%H %d %m %Y");
+// Create tooltip variables
+const tooltip = d3.select('#tooltip');
+
+// Set x and y scales
+const x = d3.scaleTime().range([0, Width]);
+const y_total_received_sms = d3.scaleLinear().range([Height, 0]);
+const y_total_sent_sms = d3.scaleLinear().range([Height, 0]);
+const y_total_failed_sms = d3.scaleLinear().range([Height, 0]);
+
+
+// Define line paths for total failed sms(s)
+const total_failed_line = d3.line()
+    .curve(d3.curveLinear)
+    .x(function (d) { return x(new Date(d.datetime)) })
+    .y(function (d) { return y_total_failed_sms(d.total_errored); })
+
+
+// Create line path element for failed line graph
+const total_failed_path = total_failed_sms_graph.append('path');
+
+// Function to update data
+const update = (data) => {
+
+    // format the data  
+    data.forEach(function (d) {
+        d.datetime = new Date(d.datetime);
+        d.total_received = +d.total_received
+        d.total_sent = +d.total_sent
+        d.total_pending = +d.total_pending
+        d.total_errored = +d.total_errored
+        d.golis_received= +d.operators["golis"]["received"]
+        d.hormud_received= +d.operators["hormud"]["received"]
+        d.nationlink_received= +d.operators["nationlink"]["received"]
+        d.somnet_received= +d.operators["somnet"]["received"]
+        d.somtel_received= +d.operators["somtel"]["received"]
+        d.telesom_received= +d.operators["telesom"]["received"]
+        d.golis_sent= +d.operators["golis"]["sent"]
+        d.hormud_sent= +d.operators["hormud"]["sent"]
+        d.nationlink_sent= +d.operators["nationlink"]["sent"]
+        d.somnet_sent= +d.operators["somnet"]["sent"]
+        d.somtel_sent= +d.operators["somtel"]["sent"]
+        d.telesom_sent= +d.operators["telesom"]["sent"]
+    });
+
+    data.sort((a, b) => new Date(a.datetime) - new Date(b.datetime));
+
+    data = data.filter(a => a.datetime > new Date("2019-07-01T12:30:00+00:00"));
```

Comment:
Was this development code? I think we should probably remove it?

Reply:
Need to limit view in for the 10min bars. Updated to render last two days.